pages/onboarding/index.tsx

```diff
@@ -1,15 +1,129 @@
-import { Box, Button, Container, Typography } from '@mui/material';
+import { Box, Button, Container, TextField, Typography } from '@mui/material';
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
 import { NextPage } from 'next';
 import Link from 'next/link';
+import { useRouter } from 'next/router';
+import React, { useState } from 'react';
+import Cookies from 'universal-cookie';
+
+import createAxiosInstance from 'client/lib/axios';
+import { UserProfile } from 'server/types/User';
+import { AUTH_COOKIE_NAME } from 'shared/constants';
+
+interface FormData {
+  bio: string;
+  twitter_username: string;
+  linkedin_url: string;
+  github_username: string;
+  website: string;
+}
+
+interface Props {
+  user: UserProfile;
+}
+
+const OnboardingPage: NextPage<Props> = ({ user }: Props) => {
+  const [formData, setFormData] = useState<FormData>({
+    bio: user.bio,
+    twitter_username: user.twitter_username,
+    linkedin_url: user.linkedin_url,
+    github_username: user.github_username,
+    website: user.website,
+  });
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isFormFilled, setIsFormFilled] = useState<boolean>(false);
+
+  const router = useRouter();
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData(prevFormData => ({ ...prevFormData, [name]: value }));
+    setIsFormFilled(true);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    try {
+      setIsLoading(true);
+      await axios.patch(`/api/users/${user.id}`, formData);
+      setIsLoading(false);
+      router.push('/directory');
+    } catch (error) {
+      setIsLoading(false);
+      console.error(error);
+    }
+  };
+
+  if (!user) {
+    router.push('/');
+  }
```

Comment:
if there was no user then the component would never get to this bit of code, it would fail at the initial `formData` state setup 
![image](https://user-images.githubusercontent.com/68509315/234386691-6d7377ec-8b8a-402a-be6c-3b267c6ef071.png)

Reply:
got it, im hoping my new patch fixes it

Reply:
This is redirecting now but since it was done in a useEffect, the page loads completely and flashes at the user before redirecting.